extensions/amp-image-slider/1.0/component.js

```diff
@@ -0,0 +1,542 @@
+          stepRight(true);
+          break;
+        case 'home':
+          e.preventDefault();
+          e.stopPropagation();
+          stepExactCenter();
+          break;
+      }
+    },
+    [animateHideHint, stepLeft, stepRight, stepExactCenter]
+  );
+  /**
+   * Handler on mouse move
+   * @param {Event} e
+   * @private
+   */
+  const onMouseMove = useCallback(
+    (e) => {
+      e.preventDefault();
+      pointerMoveX(e.pageX);
+    },
+    [pointerMoveX]
+  );
+
+  /**
+   * Handler on mouse button up
+   * @param {Event} e
+   * @private
+   */
+  const onMouseUp = useCallback(
+    (e) => {
+      e.preventDefault();
+      unlisten(unlistenMouseMove?.current);
+      unlisten(unlistenMouseUp?.current);
+    },
+    [unlisten]
+  );
+  /**
+   * Handler on mouse button down
+   * @param {Event} e
+   * @private
+   */
+  const onMouseDown = useCallback(
+    (e) => {
+      e.preventDefault();
+      pointerMoveX(e.pageX);
+
+      // In case, clear up remnants
+      // This is to prevent right mouse button down when left still down
+      unlisten(unlistenMouseMove?.current);
+      unlisten(unlistenMouseUp?.current);
+
+      unlistenMouseMove.current = listen(win.current, 'mousemove', onMouseMove);
+      unlistenMouseUp.current = listen(win.current, 'mouseup', onMouseUp);
+
+      animateHideHint();
+    },
+    [
+      animateHideHint,
+      pointerMoveX,
+      unlistenMouseMove,
+      unlistenMouseUp,
+      onMouseMove,
+      onMouseUp,
+      unlisten,
+    ]
+  );
+
+  const registerEvents = useCallback(() => {
+    if (isEventRegistered.current) {
+      return;
+    }
+    unlistenMouseDown.current = listen(
+      containerRef.current,
+      'mousedown',
+      onMouseDown
+    );
+    unlistenKeyDown.current = listen(
+      containerRef.current,
+      'keydown',
+      onKeyDown
+    );
+    registerTouchGestures();
+    //this.isEventRegistered = true;
+  }, [
+    registerTouchGestures,
+    unlistenKeyDown,
+    isEventRegistered,
+    onMouseDown,
+    onKeyDown,
+  ]);
+
+  useEffect(() => {
+    /** Common variables */
+    win.current = containerRef.current.ownerDocument.defaultView;
+    doc.current = containerRef.current.ownerDocument;
+    //setIsEdge(Services.platformFor(win).isEdge());
+
+    /** Retrieve Images */
+    images.forEach((child) => {
+      // TODO (@AnuragVasanwala): Please remove 'VALID_IMAGE_TAGNAMES' check for Preact.
+      if (VALID_IMAGE_TAGNAMES.has(child.type.toUpperCase())) {
+        if (leftImageRef.current == null) {
+          // First encountered = left image
+          //leftImageRef.current = images[0];
+        } else if (rightImageRef.current == null) {
+          // Second encountered = right image
+          //rightImageRef.current = images[1];
+          images[1].key.classList.add('i-amphtml-image-slider-push-left');
+        } else {
+          // TODO (@AnuragVasanwala): Assert Error - Should not contain more than 2 images.
+        }
+      }
+    });
```

Comment:
JFI: I am going to remove this (validation and dynamically adding of image tag).